### DISCUSSION TITLE
Allow addresses for Maps component instead of lat / long

### DISCUSSION THREAD

**Discussion by Grayda** [Budibase/budibase#5313] [Ideas]
I added a maps component to my app, but realized that it's after a latitude / longitude and doesn't allow an address. I have a Contacts table where the user's address is stored along with their name. Prior to the map component being available, I just had a link that said "Map" on `/contacts` and `/contacts/:id` that would just open a Google Maps URL with `?q=<address>` appended. Now I'm thinking of creating a static formula that calls the [Nominatim API](https://nominatim.org/release-docs/develop/api/Search/) to retrieve the lat / long, then use those columns

And on a side note, it'd be nice to be able to use one column for lat & long, instead of one for lat, one for long. The usual format is -37.828151806629386, 144.9418306314571 (or lat & long separated by a comma)

**Comment by shogunpurple** [comment]
Hi @Grayda - so our reasoning for this was exactly what you mentioned. If you need google maps, you can embed it fairly easily and pass some query parameters.

The budibase map is intended to be a much more custom, flexible option that suits internal tool use cases better because you can 
use custom tilesets, and eventually images: https://github.com/Budibase/budibase/discussions/5312

Geocoding in general is another element to the map that we will look to ship over the coming months. This will again be customisable, and you will be able to plug your maps into a whole host of different geocoders. You can set this up manually at the moment using something like Nominatim though, as you mentioned.
